### PR TITLE
InfluxDB: Incremental query PoC

### PR DESCRIPTION
**Pull request by gtk-grafana**

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 1]
```diff
@@ -166,6 +176,13 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
     }
   }
 
+  getInfluxTargetSignature(request: DataQueryRequest<InfluxQuery>, targ: InfluxQuery) {
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/influxdb/datasource.ts:179]: @itsmylife what uniquely defines a query for influx?

File: pkg/tsdb/influxdb/response_parser.go [hunk 2]
```diff
@@ -114,8 +114,10 @@ func transformRows(rows []Row, query Query) data.Frames {
 
 				for _, valuePair := range row.Values {
 					timestamp, timestampErr := parseTimestamp(valuePair[0])
+					value := valuePair[1]
+
 					// we only add this row if the timestamp is valid
-					if timestampErr == nil {
+					if timestampErr == nil && value != nil {
```
Comment (gtk-grafana) [inline pkg/tsdb/influxdb/response_parser.go:120]: So this is a huge breaking change, and probably won't fly, but this does indeed solve the problems in merging dataframes in incremental querying

Comment (gtk-grafana) [reply pkg/tsdb/influxdb/response_parser.go:120]: Also this makes the responses a good bit smaller

File: public/app/plugins/datasource/influxdb/datasource.ts (hunk 1 above)
Comment (itsmylife) [reply public/app/plugins/datasource/influxdb/datasource.ts:179]: I think we do not have a way to do this for influx.

Comment (gtk-grafana) [reply public/app/plugins/datasource/influxdb/datasource.ts:179]: In prometheus we use the query interval, the interpolated expression, exemplar status etc, I think with influx we'll just need to either echo out the interpolated query string (if raw query mode), or the JSON encoding of the select `InfluxQueryPart[][]` concatenated with the interval. Basically we're trying to figure out which properties of an influxQuery change when the query changes

Comment (gtk-grafana) [reply public/app/plugins/datasource/influxdb/datasource.ts:179]: @itsmylife I noticed that basically all of the props change the query, so I went ahead and added basically all of these props. I wasn't sure about the `adhocFilters`or `resultFormat`, do these change the backend response?

File: pkg/tsdb/influxdb/response_parser.go [hunk 3]
```diff
@@ -146,6 +146,9 @@ func transformRows(rows []Row, query Query) data.Frames {
 				name := string(formatFrameName(row, column, query, frameName[:]))
 
 				timeField := data.NewField("time", nil, timeArray)
+
+				timeField.SetConfig(&data.FieldConfig{Interval: float64(query.Interval.Milliseconds())})
```
Comment (gtk-grafana) [inline pkg/tsdb/influxdb/response_parser.go:150]: remove

**Pull request closed without merging**